bokehjs/test/integration/annotations/legend.ts

```diff
@@ -438,4 +438,56 @@ describe("Legend annotation", () => {
 
     await display(p)
   })
+
+  describe("should support as border_line_dash pattern value", () => {
+    const p = fig([200, 200])
+    const x = [1, 2, 3, 4, 5]
+    const y1 = [2, 3, 4, 5, 6]
+    const y2 = [3, 4, 5, 6, 7]
+
+    p.line(x, y1, {legend_label: "Temp.", line_color: "blue"})
+    p.line(x, y2, {legend_label: "Objects", line_color: "red"})
+
+    p.legend.location = "top_left"
+    p.legend.border_line_color = "black"
+
+    it("default", async () => {
+      await show_with_exported(p)
+    })
+
+    it("solid", async () => {
+      p.legend.border_line_dash = "solid"
```

Comment:
Using mutable state between tests is undesirable, unless this is what you want to test. Typically we would create a plot in a function, either with specific arguments or a function for modifying its contents, and then instantiate a fresh plot in every tests.


```suggestion
    function plot(dash_pattern: DashPattern) {
      const p = fig([200, 200])
      const x = [1, 2, 3, 4, 5]
      const y1 = [2, 3, 4, 5, 6]
      const y2 = [3, 4, 5, 6, 7]

      p.line(x, y1, {legend_label: "Temp.", line_color: "blue"})
      p.line(x, y2, {legend_label: "Objects", line_color: "red"})

      p.legend.location = "top_left"
      p.legend.border_line_color = "black"
      
      p.legend.border_line_dash = dash_pattern
      
      return p
    }

    it("solid", async () => {
      const p = plot("solid")
      p.legend.border_line_dash = "solid"
```
and so on.

Reply:
Also I believe testing the default in this case is a waste of test, because visual tests are very expensive and we aren't testing anything unique from the second test (from the visual perspective).